src/generatePreviewTypes.ts

```diff
@@ -0,0 +1,60 @@
+import { Property } from "./WidgetXml";
+import { capitalizeFirstLetter, extractProperties } from "./helpers";
+
+export function generatePreviewTypes(widgetName: string, properties: Property[]): string[] {
+    const results = Array.of<string>();
+    results.push(`export interface ${widgetName}PreviewProps {
+    class: string;
+    style: string;
+${generatePreviewTypeBody(properties, results)}
+}`);
+    return results;
+}
+
+function generatePreviewTypeBody(properties: Property[], generatedTypes: string[]) {
+    return properties.map((prop) => `    ${prop.$.key}: ${toPreviewPropType(prop, generatedTypes)};`).join("\n");
+}
+
+function toPreviewPropType(prop: Property, generatedTypes: string[]): string {
+    switch (prop.$.type) {
+        case "boolean":
+            return "boolean";
+        case "string":
+            return "string";
+        case "action":
+            return "{} | null";
+        case "textTemplate":
+            return "string";
+        case "integer":
+        case "decimal":
+            return "number | null";
+        case "icon":
+            return '{ type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; } | null';
+        case "image":
+            return "string";
+        case "file":
+            return "string";
+        case "datasource":
+            return "{} | null";
```

Comment:
This property has a {type: string} arg

Reply:
Yes, but then it is misaligned with what we do for actions. So I've discussed it with the team and we agreed to move towards `{} | null`. I think we can use a new type already, since the change, when rolled out, will be backwards-compatible.